rust/src/smb/smb.rs

```diff
@@ -2401,10 +2450,71 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
                 SCLogError!("Invalid value for smb.max-tx");
             }
         }
+        let retval = conf_get("app-layer.protocols.smb.max-guid-cache-size");
+        if let Some(val) = retval {
+            if let Ok(v) = val.parse::<usize>() {
+                if v > 0 {
+                    SMB_CFG_MAX_GUID_CACHE_SIZE = v;
+                } else {
+                    SCLogError!("Invalid max-guid-cache-size value");
+                }
+            } else {
+                SCLogError!("Invalid max-guid-cache-size value");
+            }
+        }
+        let retval = conf_get("app-layer.protocols.smb.max-read-offset-cache-size");
```

Comment:
Looks like some get_as_usize could be useful here :)

Reply:
Can be done as a future optimization.